config/dynamic.c

```diff
@@ -16,11 +16,535 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <sys/un.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/select.h>
+#include <lualib.h>
+#include <lauxlib.h>
+
+#include <rte_log.h>
+#include <rte_lcore.h>
+#include <rte_cycles.h>
+#include <rte_malloc.h>
+
+#include "gatekeeper_net.h"
+#include "gatekeeper_main.h"
 #include "gatekeeper_config.h"
+#include "gatekeeper_launch.h"
+
+/* TODO Get the install-path via Makefile. */
+#define LUA_DY_BASE_DIR     "./lua"
+#define DYNAMIC_CONFIG_FILE "example_of_dynamic_config_request.lua"
+
+const char *CLIENT_PROC_ERROR =
+	"function reply() return -1, \"Dynamic Configuration cannot process the request. end\"\n";
+
+/*
+ * It's because of the strict compilation check of DPDK,
+ * without uint16_t, it pops an error message
+ * "error: large integer implicitly truncated to
+ * unsigned type [-Werror=overflow]."
+ */
+const uint16_t MSG_MAX_LEN = (uint16_t)~0U;
+
+static struct dynamic_config config;
+
+/*
+ * Return values:
+ * 	0: Write @nbytes successfully.
+ * 	-1: Connection closed by the client.
+ */
+static int
+write_nbytes(int conn_fd, const char *msg_buff, int nbytes)
+{
+	int send_size;
+	int tot_size = 0;
+
+	while ((send_size = write(conn_fd, msg_buff + tot_size,
```

Comment:
Please ignore my recommendation to deal with `EINTR`. Let's keep the current behavior as a feature, that is, a way to unblock a connection. I'm sorry for the confusion.

However, do deal with the case `nbytes` is zero since it is a problem. The function can just test this at the beginning and `return 0`.